examples/carousel/esp-idf/s2-kaluga-kit/main/CMakeLists.txt

```diff
@@ -0,0 +1,20 @@
+# Copyright © SixtyFPS GmbH <info@slint.dev>
+# SPDX-License-Identifier: MIT
+
+idf_component_register(
+    SRCS "../../main.cpp"
+    INCLUDE_DIRS "."
+    REQUIRES slint
+)
+
+slint_target_sources(${COMPONENT_LIB} ../../../ui/carousel_demo.slint)
+target_link_options(${COMPONENT_LIB} PUBLIC -Wl,--allow-multiple-definition)
+set_target_properties(${COMPONENT_LIB} PROPERTIES SLINT_EMBED_RESOURCES "embed-for-software-renderer")
+
+# workaround false positive:
+# ```
+# In member function 'void slint::private_api::Repeater<...>::RepeaterInner::row_added(size_t, size_t):
+# vtable.h:111:29: error: '... vtable::VRc<...>::inner' may be used uninitialized in this function [-Werror=maybe-uninitialized]
+# 111 |     VRc(const VRc &other) : inner(other.inner) { inner->strong_ref++; }
+# ```
+target_compile_options(${COMPONENT_LIB} PRIVATE -Wno-maybe-uninitialized)
```

Comment:
This should probably also be a PUBLIC property of our slint component

Reply:
Hmm, we could do that, but that hides other uninitialised variable warnings in user code (business logic). I guess the _right_ fix is to emit pragma push/pop to disable those warnings into the generated code. What do you think?

Reply:
maybe we can add that in the vtable.h header.

I wonder if we actually do something wrong. it's hard to see because the error don't pin-point a location where the thing is declared or something

Reply:
I think it can be reproduced on the CI

Reply:
Indeed, that's using esp-idf 5.0. I have 5.1 on my machine - I'll try upgrading the CI to 5.1. Unless you think we should try to add a workaround for the old release of esp-idf?